pilot/pkg/security/authz/model/model.go

```diff
@@ -265,3 +270,27 @@ func (m *Model) Generate(service *ServiceMetadata, forTCPFilter bool) *envoy_rba
 	}
 	return policy
 }
+
+// getPrincipalsIncludingAliases returns a list of users encoded in SPIFFE format with the local
+// trust domain and its aliases.
+// For example, if the local trust domain is "cluster.local", and its aliases are "td1", "td2".
+// For a user "bar" in namespace "foo", getPrincipalsIncludingAliases returns
+// cluster.local/ns/foo/sa/bar, td1/ns/foo/sa/bar, td2/ns/foo/sa/bar
+// The trust domain corresponds to the trust root of a system.
+// Refer to
+// [SPIFFE-ID](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain)
+// In Istio, an identity is presented in the SPIFFE format, which is:
+// spiffe://<trust-domain>/ns/<some-namespace>/sa/<some-service-account>
+func getPrincipalsIncludingAliases(trustDomainAliases []string, principals []string) []string {
+	usersWithAliases := principals
+	for _, tdAlias := range trustDomainAliases {
```

Comment:
I may flip the order of these for loop to reuse the result of principal parsing (need to separate them from getIdentityWithNewTrustDomain though).

Reply:
I did that initially but I prefer the order that this looping offers. That is:
`td1/a, td1/b, td1/c, td2/a, td2/b, td2/c` vs. `td1/a, td2/a, td1/b, td2/b...`. Wdyt?